cmake/SetupCodeChecks.cmake

```diff
@@ -412,15 +453,96 @@ macro(blt_add_astyle_target)
 
     # Check/Set required parameters
     if(NOT DEFINED arg_NAME)
-        message(FATAL_ERROR "blt_add_astyle_target requires a NAME parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a NAME parameter")
     endif()
 
     if(NOT DEFINED arg_CFG_FILE)
-        message(FATAL_ERROR "blt_add_astyle_target requires a CFG_FILE parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a CFG_FILE parameter")
     endif()
 
     if(NOT DEFINED arg_SRC_FILES)
-        message(FATAL_ERROR "blt_add_astyle_target requires a SRC_FILES parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a SRC_FILES parameter")
+    endif()
+
+    if(NOT DEFINED arg_MODIFY_FILES)
+        set(arg_MODIFY_FILES FALSE)
+    endif()
+
+    if(DEFINED arg_WORKING_DIRECTORY)
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+
+    set(_generate_target TRUE)
+
+    # Copy config file to given working directory since ClangFormat doesn't support pointing to one
+    configure_file(${arg_CFG_FILE} ${arg_WORKING_DIRECTORY}/.clang-format COPYONLY)
+
+    if(_generate_target)
+        # ClangFormat does not support --dry-run until version 10 which isn't on many machines.
```

Comment:
Looks like we are always using ``run-clang-format``, even if the clang version is ``10+``

Reply:
I made a comment about it in the docs.  run-clang-format does a pretty nice job cleaning the output.